### PR TITLE
Remove binary compability job from master

### PR DESCRIPTION
**Pull request by melanieplageman**
Since ecdea364208bc899be1c7316053ab0fe78e86feb there has been a red job in the master CI https://prod.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb_master/jobs/compile_gpdb_binary_swap_centos7/builds/1 . Since a binary compatibility test for master doesn't make much sense, I thought we should remove it instead of trying to make it green.

It won't save us much work to keep it around and bump the version the binary swap test is comparing with because the binary compatibility job in 6X_STABLE works differently anyway. We are probably better off just adding it back in when we cut 7.

I set a pipeline in dev with the changes to see if they were correct
https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-remove_master_binary_swap_test

### CODE REVIEW

**Comment by jimmyyih** [comment]
+1 on the idea.

File: concourse/scripts/ic_gpdb_resgroup.bash [hunk 1]
```diff
@@ -100,54 +100,8 @@ keep_minimal_cgroup_dirs() {
 EOF
 }
 
-setup_binary_swap_test() {
```
Comment (ashwinstar) [inline concourse/scripts/ic_gpdb_resgroup.bash:103]: I don't see harm in leaving this file untouched, right? The variable will not be defined and hence the test will not run but this code can remain. When enabling the binary swap again can set the variable and test will start running again.

If code is different between 6X_STABLE for this bash file then its fine to take it out and we can bring the modified piece to master. But if this piece is same then better leave it untouched.

**Comment by tylarb** [comment]
+1 from me on this as well excepting that resource groups requires binary swap to pass. I don't know why though. If we say resource groups can run without binary swap, then please remove binary swap from master.

**Comment by melanieplageman** [comment]
> +1 from me on this as well excepting that resource groups requires binary swap to pass. I don't know why though. If we say resource groups can run without binary swap, then please remove binary swap from master.

We test binary swap for resource groups separately so the resource groups job takes the rpm that is built for the binary swap test as an input. If we don't need to run binary swap for greenplum, the same should hold for resource groups, so I removed the get for that rpm.

File: concourse/scripts/ic_gpdb_resgroup.bash (hunk 1 above)
Comment (melanieplageman) [reply concourse/scripts/ic_gpdb_resgroup.bash]: It is the same. So, I reverted it to how it was

**Pull request merged**